statemachine/statemachine.py

```diff
@@ -277,6 +277,23 @@ def __repr__(self):
             self.current_state.identifier,
         )
 
+    def _visit_neighbour_states(self, transition, visited_states):
+        for neighbour_state in transition.destinations:
+            if neighbour_state not in visited_states:
+                self._visitable_states(neighbour_state, visited_states)
+
+    def _visitable_states(self, start_state, visited_states):
+        visited_states.append(start_state)
+
+        for transition in start_state.transitions:
+            self._visit_neighbour_states(transition, visited_states)
+        return visited_states
+
+    def _is_connected(self, starting_state):
+        visitable_states = self._visitable_states(starting_state, visited_states=[])
+
+        return len(visitable_states) == len(self.states)
```

Comment:
@rafaelrds That do you think about doing an intersection like `set(self.states) - set(visitable_states)` so we can inform on the exception what are the unreachable states?